py-polars/polars/eager/series.py

```diff
@@ -720,7 +607,7 @@ def name(self) -> str:
         """
         return self._s.name()
 
-    def rename(self, name: str, in_place: bool = False) -> Optional["Series"]:
+    def rename(self, name: Optional[str], in_place: bool = False) -> Optional["Series"]:
```

Comment:
Why is the name optional? Passing a None will result in an error.

Reply:
Hm yeah. It's sort of with the idea that this should be possible in the future (see #997). I did the same in some other places (`Series._from_arrow` for example). But strictly at this point it should be typed as a `str`. I can change that if you want.

Reply:
Yes, the refactor proposed #997 is quite large and has some non trivial changes in the lazy code base, so it is not so high on my priority list. I think that after $997 we can take a look at the function that could benefit from that change.

Reply:
All right, I'll change it back and fix this issue in all other places.

Reply:
@ritchie46 Fixed now!